builder/dockerfile/evaluator_test.go

```diff
@@ -177,34 +176,23 @@ func executeTestCase(t *testing.T, testCase dispatchTestCase) {
 		t.Fatalf("Error when parsing Dockerfile: %s", err)
 	}
 
-	options := &types.ImageBuildOptions{
-		BuildArgs: make(map[string]*string),
-	}
+	n := result.AST.Children[0]
+	cmd, err := instructions.ParseCommand(n)
 
-	b := &Builder{
-		options:   options,
-		Stdout:    ioutil.Discard,
-		buildArgs: newBuildArgs(options.BuildArgs),
+	if err != nil {
+		testutil.ErrorContains(t, err, testCase.expectedError)
```

Comment:
I think, instead of the comments in the test cases above, it would be good to split this into two test cases. One for parse errors, and another for dispatch errors.